src/routes/jira/sync/jira-sync-post.ts

```diff
@@ -29,7 +30,13 @@ export const JiraSyncPost = async (req: Request, res: Response, next: NextFuncti
 			res.status(400).send("Invalid date value, cannot select a future date!");
 			return;
 		}
-		await findOrStartSync(subscription, req.log, false, syncType, commitsFromDate);
+
+		const shouldUseBackfillAlgoIncremental = await booleanFlag(BooleanFlags.USE_BACKFILL_ALGORITHM_INCREMENTAL, res.locals.installation.jiraHost);
+		if (shouldUseBackfillAlgoIncremental && isIncrementalBackfilling(subscription, commitsFromDate)) {
+			await findOrStartSync(subscription, req.log, "partial", commitsFromDate, ["pull", "branch", "commit", "build", "deployment"]);
```

Comment:
if you send no tasks it does all tasks so you can delete `["pull", "branch", "commit", "build", "deployment"]`

Reply:
+1
I think I had a discussion with @Harminder84 before and thought we need those. 
But now with a second thought, I don't remember why do we need those. Do you remember @Harminder84 ?

Reply:
At some places we're relying on tasks to clear data. Once instance is 

```
const resetTargetedTasks = async (subscription: Subscription, syncType?: SyncType, targetTasks?: TaskType[]): Promise<void> => {
	if (!targetTasks?.length) {
		return;
	}

```

Reply:
Okay, I see the reason `why` now. 
That `if (!targetTasks?.length) {	return;	}` doesn't make sense to me, but I'd agree we should leave it alone for now.